### ISSUE TITLE
crashed on tr_dh_secret_derived

### ISSUE THREAD

**Issue by zer09** [transmission/transmission#380]
I am trying to download [sdi-tool](https://sdi-tool.org/) but after adding the torrent, transmission will crash.
I also have a crash report with a core dump.
```

Process 7180 (transmission-gt) of user 1000 dumped core.

Stack trace of thread 7187:
#0  0x00000015c6918aa3 tr_dh_secret_derive (transmission-gtk)
#1  0x00000015c693bbb8 canRead (transmission-gtk)
#2  0x00000015c69278a2 canReadWrapper (transmission-gtk)
#3  0x00000015c694d260 _Z19UTP_ProcessIncomingP9UTPSocketPKhmb (transmission-gtk)
#4  0x00000015c694dcfc UTP_IsIncomingUTP (transmission-gtk)
#5  0x00000015c690e2cc tr_utpPacket (transmission-gtk)
#6  0x00000015c690da99 event_callback (transmission-gtk)
#7  0x00007fc690cdb3f9 event_base_loop (libevent-2.0.so.5)
#8  0x00000015c690e5d8 libeventThreadFunc (transmission-gtk)
#9  0x00000015c68f925a ThreadFunc (transmission-gtk)
#10 0x00007fc68fc3936d start_thread (libpthread.so.0)
#11 0x00007fc68f971bbf __clone (libc.so.6)

Stack trace of thread 7180:
#0  0x00007fc68f965acd poll (libc.so.6)
#1  0x00007fc690f5e569 g_main_context_iterate.isra.25 (libglib-2.0.so.0)
#2  0x00007fc690f5e67c g_main_context_iteration (libglib-2.0.so.0)
#3  0x00007fc69191debd g_application_run (libgio-2.0.so.0)
#4  0x00000015c68d4800 main (transmission-gtk)
#5  0x00007fc68f88150a __libc_start_main (libc.so.6)
#6  0x00000015c68d4a2a _start (transmission-gtk)

Stack trace of thread 7182:
#0  0x00007fc68f965acd poll (libc.so.6)
#1  0x00007fc690f5e569 g_main_context_iterate.isra.25 (libglib-2.0.so.0)
#2  0x00007fc690f5e67c g_main_context_iteration (libglib-2.0.so.0)
#3  0x00007fc690f5e6c1 glib_worker_main (libglib-2.0.so.0)
#4  0x00007fc690f85536 g_thread_proxy (libglib-2.0.so.0)
#5  0x00007fc68fc3936d start_thread (libpthread.so.0)
#6  0x00007fc68f971bbf __clone (libc.so.6)

Stack trace of thread 7188:
#0  0x00007fc68fc4383d __nanosleep (libpthread.so.0)
#1  0x00000015c690f6f6 tr_wait_msec (transmission-gtk)
#2  0x00000015c690c28f dht_bootstrap (transmission-gtk)
#3  0x00000015c68f925a ThreadFunc (transmission-gtk)
#4  0x00007fc68fc3936d start_thread (libpthread.so.0)
#5  0x00007fc68f971bbf __clone (libc.so.6)

Stack trace of thread 7266:
#0  0x00007fc68f96c559 syscall (libc.so.6)
#1  0x00007fc690fa36fa g_cond_wait_until (libglib-2.0.so.0)
#2  0x00007fc690f32b31 g_async_queue_pop_intern_unlocked (libglib-2.0.so.0)
#3  0x00007fc690f330ec g_async_queue_timeout_pop (libglib-2.0.so.0)
#4  0x00007fc690f85fde g_thread_pool_thread_proxy (libglib-2.0.so.0)
#5  0x00007fc690f85536 g_thread_proxy (libglib-2.0.so.0)
#6  0x00007fc68fc3936d start_thread (libpthread.so.0)
#7  0x00007fc68f971bbf __clone (libc.so.6)

Stack trace of thread 7371:
#0  0x00007fc68f9611cd __read (libc.so.6)
#1  0x00007fc68f8e1198 _IO_file_underflow@@GLIBC_2.2.5 (libc.so.6)
#2  0x00007fc68f8e2442 _IO_default_uflow (libc.so.6)
#3  0x00007fc68f8d40c2 _IO_getline (libc.so.6)
#4  0x00007fc68f8d2f9d _IO_fgets (libc.so.6)
#5  0x00007fc65c5f6e7a avahi_resolve_name (libnss_mdns4_minimal.so.2)
#6  0x00007fc65c5f7260 _nss_mdns4_minimal_gethostbyname2_r (libnss_mdns4_minimal.so.2)
#7  0x00007fc68f957c64 gaih_inet.constprop.5 (libc.so.6)
#8  0x00007fc68f958f8d getaddrinfo (libc.so.6)
#9  0x00007fc690a91ed7 Curl_getaddrinfo_ex (libcurl.so.4)
#10 0x00007fc690a9c7af getaddrinfo_thread (libcurl.so.4)
#11 0x00007fc690a9a16b curl_thread_create_thunk (libcurl.so.4)
#12 0x00007fc68fc3936d start_thread (libpthread.so.0)
#13 0x00007fc68f971bbf __clone (libc.so.6)

Stack trace of thread 7372:
#0  0x00007fc68f9611cd __read (libc.so.6)
#1  0x00007fc68f8e1198 _IO_file_underflow@@GLIBC_2.2.5 (libc.so.6)
#2  0x00007fc68f8e2442 _IO_default_uflow (libc.so.6)
#3  0x00007fc68f8d40c2 _IO_getline (libc.so.6)
#4  0x00007fc68f8d2f9d _IO_fgets (libc.so.6)
#5  0x00007fc65c5f6e7a avahi_resolve_name (libnss_mdns4_minimal.so.2)
#6  0x00007fc65c5f7260 _nss_mdns4_minimal_gethostbyname2_r (libnss_mdns4_minimal.so.2)
#7  0x00007fc68f957c64 gaih_inet.constprop.5 (libc.so.6)
#8  0x00007fc68f958f8d getaddrinfo (libc.so.6)
#9  0x00007fc690a91ed7 Curl_getaddrinfo_ex (libcurl.so.4)
#10 0x00007fc690a9c7af getaddrinfo_thread (libcurl.so.4)
#11 0x00007fc690a9a16b curl_thread_create_thunk (libcurl.so.4)
#12 0x00007fc68fc3936d start_thread (libpthread.so.0)
#13 0x00007fc68f971bbf __clone (libc.so.6)

Stack trace of thread 7380:
#0  0x00007fc68f9611cd __read (libc.so.6)
#1  0x00007fc68f8e1198 _IO_file_underflow@@GLIBC_2.2.5 (libc.so.6)
#2  0x00007fc68f8e2442 _IO_default_uflow (libc.so.6)
#3  0x00007fc68f8d40c2 _IO_getline (libc.so.6)
#4  0x00007fc68f8d2f9d _IO_fgets (libc.so.6)
#5  0x00007fc65c5f6e7a avahi_resolve_name (libnss_mdns4_minimal.so.2)
#6  0x00007fc65c5f7260 _nss_mdns4_minimal_gethostbyname2_r (libnss_mdns4_minimal.so.2)
#7  0x00007fc68f957c64 gaih_inet.constprop.5 (libc.so.6)
#8  0x00007fc68f958f8d getaddrinfo (libc.so.6)
#9  0x00007fc690a91ed7 Curl_getaddrinfo_ex (libcurl.so.4)
#10 0x00007fc690a9c7af getaddrinfo_thread (libcurl.so.4)
#11 0x00007fc690a9a16b curl_thread_create_thunk (libcurl.so.4)
#12 0x00007fc68fc3936d start_thread (libpthread.so.0)
#13 0x00007fc68f971bbf __clone (libc.so.6)

Stack trace of thread 7184:
#0  0x00007fc68f965acd poll (libc.so.6)
#1  0x00007fc690f5e569 g_main_context_iterate.isra.25 (libglib-2.0.so.0)
#2  0x00007fc690f5e902 g_main_loop_run (libglib-2.0.so.0)
#3  0x00007fc691949cb6 gdbus_shared_thread_func (libgio-2.0.so.0)
#4  0x00007fc690f85536 g_thread_proxy (libglib-2.0.so.0)
#5  0x00007fc68fc3936d start_thread (libpthread.so.0)
#6  0x00007fc68f971bbf __clone (libc.so.6)

Stack trace of thread 7189:
#0  0x00007fc68f967d03 __select (libc.so.6)
#1  0x00000015c6917256 tr_webThreadFunc (transmission-gtk)
#2  0x00000015c68f925a ThreadFunc (transmission-gtk)
#3  0x00007fc68fc3936d start_thread (libpthread.so.0)
#4  0x00007fc68f971bbf __clone (libc.so.6)

Stack trace of thread 7185:
#0  0x00007fc68f965acd poll (libc.so.6)
#1  0x00007fc690f5e569 g_main_context_iterate.isra.25 (libglib-2.0.so.0)
#2  0x00007fc690f5e67c g_main_context_iteration (libglib-2.0.so.0)
#3  0x00007fc67c672f3d dconf_gdbus_worker_thread (libdconfsettings.so)
#4  0x00007fc690f85536 g_thread_proxy (libglib-2.0.so.0)
#5  0x00007fc68fc3936d start_thread (libpthread.so.0)
#6  0x00007fc68f971bbf __clone (libc.so.6)
```
OS: fedora 26 kernel 4.12.5
Transmission version: 2.92 (14714)

thanks

**Comment by mikedld** [comment]
I wasn't able to reproduce so please upload the core dump somewhere.

**Comment by mikedld** [comment]
As it could contain sensitive information, you could drop me a mail at mike@transmissionbt.com instead of linking here.

**Comment by zer09** [comment]
Thanks, email sent.

**Comment by mikedld** [comment]
Already fixed in #27.
